docs/experiments/extract-mode.md

```diff
@@ -0,0 +1,103 @@
+---
+id: extract-mode
+append_help_link: true
+description: "This article explains the extract mode, which allows for easier handling of files containing more than one language."
+---
+
+# Extract mode
+
+## Introduction
+
+Extract mode enables you to run existing rules on subsections of files where the rule language is different than the language of the file. For example, running a Javascript rule on code contained inside of script tags in an HTML document.
+
+## Example of extract mode
+
+Without extract mode, writing rules to validate template, Markdown or configuration files which contain code in another language can be burdensome and require significant rule duplication.
+
+Let's take the following Bash rule as an example (a simplified version of [our `curl-eval` rule](https://github.com/returntocorp/semgrep-rules/blob/release/bash/curl/security/curl-eval.yaml)):
+
+```yaml
+rules:
+  - id: curl-eval
+    severity: WARNING
+    languages:
+      - bash
+    message: Evaluating data from a `curl` command is unsafe.
+    mode: taint
+    pattern-sources:
+      - pattern: |
+          $(curl ...)
+      - pattern: |
+          `curl ...`
+    pattern-sinks:
+      - pattern: eval ...
+```
+
+Usually, this rule would only be run against Bash files. However, our project might contain Dockerfiles or Python scripts which invoke Bash commands&mdash;this rule won't currently be run against any Bash contained in those files.
+
+However, with extract mode we can supplement our Bash rule with instructions on how to extract any Bash commands we might be using in a Docker `RUN` instruction or via Python's `os.system` standard library function.
```

Comment:
```suggestion
However, with extract mode we can supplement our Bash rule with instructions on how to extract any Bash commands we might be using in a Docker `RUN` instruction or through Python's `os.system` standard library function.
```

Reply:
Avoid this abbreviation and use through more often. It can be a bit harder to understand by non-natives (see via in https://developers.google.com/style/word-list?hl=en).